solutions/she/lab3/kern/pmap.c

```diff
@@ -227,6 +252,17 @@ mem_init(void)
 	check_page_installed_pgdir();
 }
 
+bool
+page_allocated_during_boot(size_t page_num) {
+	char *l_bound = (char *) KADDR((page_num * PGSIZE));
+	char *r_bound = (char *) KADDR(((page_num + 1) * PGSIZE - 1)) + 1;
+	// if (page_num < 160) {
```

Comment:
Are there some other ways to calculate these upper and lower boundaries?

Reply:
Well... Yes, probably. Why are you asking? I use this way because I don't want to show that I know behavior of KADDR function. In my imlementation one can change segmentation mapping by changing KADDR, but still have this code working